pkg/build/prometheus/metrics.go

```diff
@@ -0,0 +1,109 @@
+package prometheus
+
+// NOTE, this module explicitly attempts to avoid references to the build api types/structs
+// in an attempt to start breaking up our compile dependencies
+
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// BuildSubsystem is how this build controller is represented in
+// prometheus metrics.
+const BuildSubsystem = "build_controller"
+
+var (
+	// decided not to have a separate counter for failed builds, which have reasons,
+	// vs. the other "finished" builds phases, where the reason is not set
+	buildsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: BuildSubsystem,
+			Name:      "build_counters",
+			Help:      "Counts total teriminal builds by phase, and in case of failures, reason",
+		},
+		[]string{"phase", "reason"},
+	)
+	buildsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: BuildSubsystem,
+			Name:      "build_gauges",
+			Help:      "Show current counts of non-terminal builds by phase",
+		},
+		[]string{"phase"},
+	)
+	buildsOverallHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: BuildSubsystem,
+			Name:      "build_timing_histograms",
+			Help:      "Histogram of durations of builds in seconds that have exited by phase",
```

Comment:
see above comment re: labels can distinguish in the prometheus console vs. separate metrics ... from the console usage I've done so far ... the use of labels to distinguish vs. separate metrics seems cleaner